validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -163,7 +187,22 @@ public void setFeeRecipient(final BLSPublicKey publicKey, final Eth1Address eth1
       throw new SetFeeRecipientException(
           "Validator public key has been configured in validators-proposer-config file - cannot update via api.");
     }
-    runtimeProposerConfig.addOrUpdate(publicKey, eth1Address);
+    runtimeProposerConfig.addOrUpdateFeeRecipient(publicKey, eth1Address);
+  }
+
+  public void setGasLimit(final BLSPublicKey publicKey, final UInt64 gasLimit)
+      throws SetFeeRecipientException {
+    if (validatorIndexCannotBeResolved(publicKey)) {
+      throw new SetGasLimitException(
+          "Validator public key not found when attempting to set gas limit.");
+    }
+    Optional<UInt64> maybeGasLimit =
+        maybeProposerConfig.flatMap(config -> getGasLimitFromProposerConfig(config, publicKey));
+    if (maybeGasLimit.isPresent()) {
+      throw new SetGasLimitException(
+          "Validator public key has been configured in validators-proposer-config file - cannot update via api.");
```

Comment:
should probably make these messages 'cannot update gas limit via api.', and maybe update the first message as well (for fee recipient)